src/Util/Config.php

```diff
@@ -138,28 +157,56 @@ public function getApiUrl(): ?string
     }
 
     /**
-     * @param array<string, mixed> $options
+     * @param mixed $client
      */
-    private function setHttpClient(array $options): void
+    private function setHttpClient($client): void
     {
-        if (isset($options[self::HTTP_CLIENT])) {
-            $value = $options[self::HTTP_CLIENT];
-            if (!($value instanceof ClientInterface)) {
-                throw new \InvalidArgumentException(sprintf(
-                    '%s configuration value must be of type %s',
-                    self::HTTP_CLIENT,
-                    ClientInterface::class
-                ));
-            }
-            $this->set(self::HTTP_CLIENT, $value);
+        if (!($client instanceof ClientInterface)) {
+            throw new \InvalidArgumentException(sprintf(
+                self::TYPE_ERROR_MESSAGE,
+                self::HTTP_CLIENT,
+                ClientInterface::class
+            ));
         }
+
+        $this->set(self::HTTP_CLIENT, $client);
     }
 
     /**
-     * @return \Psr\Http\Client\ClientInterface|null
+     * @return \Psr\Http\Client\ClientInterface
      */
-    public function getHttpClient(): ?ClientInterface
+    public function getHttpClient(): ClientInterface
     {
+        if ($this->get(self::HTTP_CLIENT) === null) {
+            $this->set(self::HTTP_CLIENT, new Client());
```

Comment:
Creates a default HTTP client lazily. This benefits the request builder when configured with a custom HTTP client.